src/shadows.cpp

```diff
@@ -0,0 +1,51 @@
+#include "aldente.h"
+#include "shadows.h"
+#include "shaders/shader_manager.h"
+#include "shaders/shadow_shader.h"
+#include "util/util.h"
+#include "window.h"
+#include "util/config.h"
+
+Shadows *Shadows::shadows = new Shadows();
+
+Shadows::Shadows() {}
+Shadows::~Shadows() {}
+
+void Shadows::shadow_pass(Scene* scene)
+{
+	float far_plane;
+	Config::config->get_value(Config::str_far_plane, far_plane);
+
+	scene->camera->update_frustum_planes();
+	scene->camera->update_frustum_corners(Window::window->width, Window::window->height, far_plane);
+
+	ShadowShader * ss = (ShadowShader *)ShaderManager::get_shader_program("shadow");
+	glViewport(0, 0, ss->size, ss->size);
+	glBindFramebuffer(GL_FRAMEBUFFER, ss->FBO);
+	glClear(GL_DEPTH_BUFFER_BIT);
+	ss->use();
+	ss->light_pos = scene->light_pos;
+	if (Aldente::aldente->shadows_on)
```

Comment:
I think I see why we're using Aldente as a global static instance... if it's for config stuff like this, I think it's better to have some static config class you can include.  As it is, `#include "aldente.h"` can potentially lead to a circular dependency down the line